core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/text/TextSubstring.java

```diff
@@ -46,6 +50,18 @@ private void verifyArguments(String text, int start, int end) {
         }
     }
 
+
+    public String substring(String string, int beginIndex, int endIndex){
+            return string.substring(beginIndex-1,endIndex);
+    }
+
+    public String substring(String string, int beginIndex){
+            return string.substring(beginIndex-1);
+    }
+
+
+
+
```

Comment:
An second pattern like the one on line 18 (INPUT_PARAMETER_PATTERN) should be created. This pattern should allow for two arguments. If the first pattern does not match the provided string, the second pattern should be checked. If none of the patterns match the Exception on line 25 should be thrown. Should one of the patterns match, the appropriate substring method should be called.

Reply:
Ok, I'll add this pattern ASAP, thanks